skcosmo/selection/FPS.py

```diff
@@ -51,16 +52,42 @@ class _BaseFPS:
                  is random
     :type idxs: list of int, None
 
+    :param progress_bar: option to use `tqdm <https://tqdm.github.io/>`_
+                         progress bar to monitor selections
+    :type progress_bar: boolean
+
     """
 
-    def __init__(self, tol=1e-12, idxs=None):
+    def __init__(self, tol=1e-12, idxs=None, progress_bar=False):
         if idxs is not None:
             self.idx = idxs
         else:
             self.idx = [np.random.randint(self.product.shape[0])]
         self.distances = np.min([self.calc_distance(i) for i in self.idx], axis=0)
         self.distance_selected = np.nan * np.zeros(self.product.shape[0])
 
+        if progress_bar:
+            try:
+                # first check whether or not this is being called from a jupyter
+                # notebook
+                try:
+                    cfg = get_ipython().config
+                    if cfg["IPKernelApp"]["parent_appname"] == "ipython-notebook":
+                        from tqdm.notebook import tqdm
+                    else:
+                        from tqdm import tqdm
+                except NameError:
+                    from tqdm import tqdm
+                self.pw = tqdm
+            except ImportError:
+                raise ImportError(
+                    "tqdm must be installed to use a progress bar."
+                    "Either install tqdm or re-run with"
+                    "progress_bar = False"
+                )
```

Comment:
This is the same code as in CUR.py, why not extract it as a separate function?

Reply:
I've put it in the `utils` folder, I anticipate this isn't the only package which might use a progress bar.